aws/resource_aws_budgets_budget.go

```diff
@@ -226,9 +331,77 @@ func resourceAwsBudgetsBudgetRead(d *schema.ResourceData, meta interface{}) erro
 
 	d.Set("time_unit", budget.TimeUnit)
 
+	return resourceAwsBudgetsBudgetNotificationRead(d, meta)
+}
+
+func resourceAwsBudgetsBudgetNotificationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+
+	accountID, budgetName, err := decodeBudgetsBudgetID(d.Id())
+
+	describeNotificationsForBudgetOutput, err := client.DescribeNotificationsForBudget(&budgets.DescribeNotificationsForBudgetInput{
+		BudgetName: aws.String(budgetName),
+		AccountId:  aws.String(accountID),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	notifications := make([]map[string]interface{}, 0)
+
+	for _, notificationOutput := range describeNotificationsForBudgetOutput.Notifications {
+		setNotificationDefaults(notificationOutput)
+		notification := make(map[string]interface{})
+		notification["comparison_operator"] = *notificationOutput.ComparisonOperator
```

Comment:
To prevent potential panics, we should prefer to use the [AWS Go SDK provided conversion functions](https://docs.aws.amazon.com/sdk-for-go/api/aws/) here and below, e.g.

```suggestion
		notification["comparison_operator"] = aws.StringValue(notificationOutput.ComparisonOperator)
```